_health-care/_js/_components/full_name.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import _ from 'lodash';
+
+class FullName extends React.Component {
+  constructor() {
+    super();
+    this.state = {hasError: false};
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillMount() {
+    this.id = _.uniqueId('date-input-');
+  }
+
+  handleChange() {
+    const name = {
+      first: this.refs.first.value,
+      middle: this.refs.middle.value,
+      last: this.refs.last.value,
+      suffix: this.refs.suffix.value
+    }
+
+    if (!this.validate(name.first, name.last)) {
+      this.setState({hasError: true});
+    } else {
+      this.setState({hasError: false});
+    }
+
+    this.props.onUserInput(name);
```

Comment:
One thing my friend warned me about in React is that you might notice a performance hit if you update things too frequently. One way we could avoid this is to only update if validation passes. Here we're updating regardless of the outcome of validation.